feat: Compute sender verification key to send to outbound dispatcher

talwinder50 opened the pull request:
closes #299

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

fqutishat commented on pkg/didcomm/protocol/didexchange/states.go:
```diff
@@ -468,7 +479,22 @@ func (ctx *context) handleInboundResponse(response *Response) error {
 }
+
+func (ctx *context) createSendVerKey() (string, error) {
+	didDoc, err := ctx.didWallet.CreateDID(didMethod)
```
you should use the same did for populating exchange request

talwinder50 replied:
done

talwinder50 commented:
unit-test

llorllale commented on pkg/didcomm/protocol/didexchange/states_test.go:
```diff
@@ -435,113 +481,16 @@ func TestCompletedState_Execute(t *testing.T) {
 	})
-}
-
-func TestCreateAndSendRequest(t *testing.T) {
```
why is this removed?

llorllale commented on pkg/didcomm/protocol/didexchange/states_test.go:
```diff
@@ -435,113 +481,16 @@ func TestCompletedState_Execute(t *testing.T) {
-	})
-}
-
-func TestCreateAndSendResponse(t *testing.T) {
```
why is this removed?

talwinder50 commented:
Since i removed small helper function and refactored execute main handling function for states. These tests are covered under other test fuction and no longer required. The file testing coverage is also above 85
